src/components/Components/OpenSimSkySphere.tsx

```diff
@@ -1,37 +1,33 @@
-import { useTexture } from '@react-three/drei';
-import { useRef } from 'react';
-import * as THREE from 'three';
+import { useLoader } from '@react-three/fiber';
 import { observer } from 'mobx-react';
+import { useRef } from 'react';
+import { Mesh, TextureLoader } from 'three';
 import { useModelContext } from '../../state/ModelUIStateContext';
 
 interface SkySphereProps {
-  texturePath?: string;
+  texturePath: string;
 }
 
-const SkySphere: React.FC<SkySphereProps> = observer(({ texturePath }) => {
-  // Load 360 sky texture
-  const curState = useModelContext();
-  const viewerState = curState.viewerState
-  const skyTexture = useTexture(viewerState.defaultSkyTextures[viewerState.skyTextureIndex]);
-  const skySphereRef = useRef<THREE.Mesh>(null);
-
-  const skyGeometry = new THREE.SphereGeometry(50, 60, 40);
-  const skyMaterial = new THREE.MeshBasicMaterial({
-    map: skyTexture,
-    side: THREE.BackSide,
-    depthWrite: false,
-  });
+const SkySphere = ({ texturePath }: SkySphereProps) => {
+  const viewerState = useModelContext().viewerState;
+  const skyTexture = useLoader(TextureLoader, viewerState.defaultSkyTextures[viewerState.skyTextureIndex]);
+  const skySphereRef = useRef<Mesh>(null);
```

Comment:
How does this handle the case where there's no texture and there's a background color or is that guarded against upstream?

Reply:
At this moment the SkySphere does not have an option to have a background color, only a texture.

I see two options here:
1. Allowing selecting a plain color forthe skysphere instead of textures, this would be a new option for the skysphere in the tree, compiling all options for background into the same space. In this case, the skysphere will always be visible.
2. Allowing making the skysphere not visible (basically removing it), and using scene/renderer background color propierties. But I am not sure where this option could live.

I think option 1 would be easier to implement and more clear, as everything related to backgrounds would be located in this class.

Reply:
I agree that option 1 would be easier but I don't know if it will allow us to create images with transparent background. If yes then I'd go for it, if not we'd need two separate options as in 4.5 where scene has a color background that would be used unless a skysphere/skybox is visible.